src/compiler/commandLineParser.ts

```diff
@@ -1192,6 +1192,14 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
         category: Diagnostics.Modules,
         description: Diagnostics.Conditions_to_set_in_addition_to_the_resolver_specific_defaults_when_resolving_imports,
     },
+    {
+        name: "resolveSideEffectImports",
+        type: "boolean",
+        affectsModuleResolution: true,

```

Comment:
i dont think you want `affecsModuleResolution` as resolution isnt changing - that is whether and how we resolve that import doesnt change? Its `affectsSemanticDiagnostics` ?

Reply:
Ah, true; I'm not 100% certain what each of them "mean" in total. I think I just repeated the same options as for `json` but that's not the same situation at all.

Reply:
Changed it, though no test changed, odd.

Reply:
You will see that only if you add test in tsc-watch or tsserver where the option changes between edits. not till then.